core/src/main/java/io/questdb/cairo/wal/WalWriter.java

```diff
@@ -1333,20 +1366,35 @@ private void switchColumnsToNewSegment(LongList newColumnFiles) {
                 long currentOffset = newColumnFiles.get(i * NEW_COL_RECORD_SIZE + 1);
                 long newOffset = newColumnFiles.get(i * NEW_COL_RECORD_SIZE + 2);
                 primaryColumnFile.jumpTo(currentOffset);
+                primaryColumnFile.sync(false);
```

Comment:
This call should use commit mode from config just like the below `secondaryColumnFile.sync(false)` call.

Reply:
fixed here https://github.com/questdb/questdb/pull/3304/commits/35152031def614e94ef1d37c6cb20f56749660c9